merge/update.go

```diff
@@ -34,6 +36,14 @@ type Updater struct {
 	IgnoredFields map[fieldpath.APIVersion]*fieldpath.Set
 
 	enableUnions bool
+
+	// Stop comparing the new object with old object after applying.
+	// This was initially used to avoid spurious etcd update, but
+	// since that's vastly inefficient, we've come-up with a better
+	// way of doing that. Create this flag to stop it.
+	// Comparing has become more expensive too now that we're not using
+	// `Compare` but `value.Equals` so this gives an option to avoid it.
+	StopComparingAfterApply atomic.Bool
```

Comment:
Hate to nit on naming but I think something like `ApplyAllowUnchangedResult` is more apt

Reply:
Thanks for nitting on this actually, I hate the name I was just trying to make you have an opinion, it worked :-)

`ApplyAllowUnchangedResult` is that accurate? Both allow unchanged result? `ReturnNilOnApplyNoop`? Also it has to be enabled by default so.

Reply:
`ApplyNoopReturnsInput`?